### PR TITLE
feat(eos_designs): Adding MACSEC and PTP to L3edge

### LINKED ISSUE #731: Add mac security, ptp to l3_edge feature
**Issue Type**

- [ ] Template enhancement
- [x] Role enhancement
- [ ] Documentation enhancement (avd.sh)

**Is your feature request related to a problem? Please describe.**

Similar to qos profile we will need to add under interface the option to have mac security profile or ptp enable.

```
  p2p_links:
      # Unique id per subnet_summary. Used to calculate ip addresses | Required with ip_pool
    - id: < integer - starting from 1 >
      # MAC sec Profile | Optional
      macsec_profile: < macsec_profile_name >
      # PTP | Optional
      ptp_enable: < p2p_profile_name >
```

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by xaviramon**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [x] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Fixes issue #731 

## Component(s) name

`arista.avd.eos_l3ls_evpn`

## Proposed changes

Changes under logic template switch-l3-edge and interface template switch-l3-edge-p2p-interfaces to read new variables from L3-edge abstraction and generate the YAML lines needed for eos_cli_config_gen to create the EOS CLI lines to use the feature.
The MACSEC profile and the PTP configuration are expected to be already set up in the device. This changes just modify enablement under defined interfaces.

## How to test

Molecule test is included, under TWODC_5STAGE_CLOS scenario.

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://www.avd.sh/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

**Review by onurgashi — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md [hunk 1]
```diff
@@ -926,6 +926,12 @@ l3_edge:
       # QOS Service Profile | Optional
       qos_profile: < qos_profile_name >
 
+      # MACSec Profile | Optional
+      macsec_profile: < macsec_profile_name >
```
Comment (titom73) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md]: Seems MACSEC profile definition is not in readme file

File: ansible_collections/arista/avd/molecule/eos_l3ls_evpn-twodc-5stage-clos/inventory/group_vars/TWODC_5STAGE_CLOS.yml [hunk 2]
```diff
@@ -47,6 +48,7 @@ l3_edge:
       ip_pool: pool-super-spine
       profile: generic-profile
       include_in_underlay_protocol: true
+      macsec_profile: MACSEC_PROFILE
```
Comment (titom73) [inline ansible_collections/arista/avd/molecule/eos_l3ls_evpn-twodc-5stage-clos/inventory/group_vars/TWODC_5STAGE_CLOS.yml]: How should we configure MACSEC profile?

**Review by titom73 — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_l3ls_evpn-twodc-5stage-clos/inventory/group_vars/TWODC_5STAGE_CLOS.yml (hunk 2 above)
Comment (onurgashi) [reply ansible_collections/arista/avd/molecule/eos_l3ls_evpn-twodc-5stage-clos/inventory/group_vars/TWODC_5STAGE_CLOS.yml]: It should be a string that is used when you generate it either via cli_config_gen or manually on the switch.
The idea is just to apply that profile to the interface.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md]: @titom73 The actual `macsec_profiles` are only available in `eos_cli_config_gen` data model, but here we just need to reference it. There is currently no need to abstract `macsec_profiles` in `eos_design`.

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md [hunk 3]
```diff
@@ -1357,6 +1363,7 @@ port_profiles:
     flowcontrol:
       received: < received | send | on >
     qos_profile: < qos_profile_name >
+    macsec_profile: < macsec_profile_name >
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md]: @xaviramon This is not added to any templates, so I have removed it from the rebase.

**Comment by ClausHolbechArista** [comment]
Rebased to latest devel

File: ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md (hunk 1 above)
Comment (titom73) [reply ansible_collections/arista/avd/roles/eos_l3ls_evpn/README.md]: Ack thanks

**Review by titom73 — APPROVED** [review]
LGTM!

**Pull request merged**